app/src/androidMain/kotlin/de/westnordost/streetcomplete/quests/ferry/AddFerryAccessMotorVehicle.kt

```diff
@@ -26,4 +31,44 @@ class AddFerryAccessMotorVehicle : OsmFilterQuestType<Boolean>(), AndroidQuest {
     override fun applyAnswerTo(answer: Boolean, tags: Tags, geometry: ElementGeometry, timestampEdited: Long) {
         tags["motor_vehicle"] = answer.toYesNo()
     }
+
+    override fun getApplicableElements(mapData: MapDataWithGeometry): Iterable<Element> {
+        // This is the primary method that performs the full check with mapData
+        return mapData.ways.filter { element ->
+            // First, a quick check with the preliminary filter
+            if (!preliminaryFilter.matches(element)) return@filter false
+            // Then, the detailed check which includes relation checks
+            isApplicableTo(element, mapData)
+        }
+    }
+
+    override fun isApplicableTo(element: Element): Boolean? {
+        // Check if it's a ferry. If yes, check if its part of a ferry route via getApplicableElements.
+        val tags = element.tags
+        if (tags["route"] == "ferry") {
+            return null
+        }
+
+        return true
+    }
+
+    private fun isApplicableTo(element: Element, mapData: MapDataWithGeometry): Boolean {
+        val tags = element.tags
+
+        // Filter out ferries that are part of a ferry route relation
```

Comment:
maybe it is possible to extract this to shared `de.westnordost.streetcomplete.quests.ferry` file? 

Like was done in say https://github.com/streetcomplete/StreetComplete/blob/master/app/src/androidMain/kotlin/de/westnordost/streetcomplete/quests/construction/MarkCompletedConstructionUtils.kt ?

(note: I have lower confidence here, maybe wait if someone else will protest)